core/src/main/java/org/infinispan/container/DataContainer.java

```diff
@@ -160,9 +171,23 @@
     * @param action the specified action to execute on filtered key/values
     * @throws InterruptedException
     */
-   public void executeTask(KeyValueFilter<? super K, ? super V> filter, BiConsumer<? super K, InternalCacheEntry<K, V>> action) throws InterruptedException;
+   void executeTask(KeyValueFilter<? super K, ? super V> filter, BiConsumer<? super K, InternalCacheEntry<K, V>> action) throws InterruptedException;
+
+   /**
+    * {@inheritDoc}
+    * <p>This iterator only returns entries that are not expired, however it will not remove them while doing so.</p>
+    * @return iterator that doesn't produce expired entries
+    */
+   @Override
+   Iterator<InternalCacheEntry<K, V>> iterator();
+
+   /**
+    * Same as {@link DataContainer#iterator()} except that is also returns expired entries.
+    * @return iterator that returns all entries including expired ones
+    */
+   Iterator<InternalCacheEntry<K, V>> expiredIterator();
```

Comment:
Sure I was debating which way I should do it, was hoping someone else would chime in :+1: